test/it/uk/gov/hmrc/component/CaseSpec.scala

```diff
@@ -469,11 +513,158 @@ class CaseSpec extends BaseFeatureSpec {
+    }
+
+    scenario("filtering by existing commodity code") {
+
+      storeCases(c1, c2, c5, c6)
+
+      val result = Http(s"$serviceUrl/cases?commodity_code=12345678").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c2,c6))
+    }
+
+    scenario("Starts-with match") {
+
+      storeCases(c1, c2, c5, c6)
+
+      val result = Http(s"$serviceUrl/cases?commodity_code=123").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c2,c6))
+    }
+
+    scenario("Contains-match does not return any result") {
+
+      storeCases(c2, c6)
+
+      val result = Http(s"$serviceUrl/cases?commodity_code=456").asString
+
+      result.code shouldEqual OK
+      result.body.toString shouldBe "[]"
+    }
+
+  }
+
+
+  feature("Get Cases by good description") {
+
+    scenario("No matches") {
+
+      storeCases(c1, c2, c5)
+
+      val result = Http(s"$serviceUrl/cases?good_description=laptop").asString
+
+      result.code shouldEqual OK
+      result.body.toString shouldBe "[]"
+    }
+
+    scenario("Filtering by existing good description") {
+
+      storeCases(c1, c2, c7)
+
+      val result = Http(s"$serviceUrl/cases?good_description=LAPTOP").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c7))
+    }
+
+    scenario("Case-insensitive search") {
+
+      storeCases(c1, c2, c7)
+
+      val result = Http(s"$serviceUrl/cases?good_description=laptop").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c7))
+    }
+
+    scenario("Filtering by substring") {
+
+      storeCases(c1, c2, c7, c8)
+
+      val result = Http(s"$serviceUrl/cases?good_description=laptop").asString
+
+      result.code shouldEqual OK
+      Json.parse(result.body) shouldBe Json.toJson(Seq(c7, c8))
+    }
+  }
+
+
+  feature("Get Cases by multiple parameters") {
+    // TODO
+  }
+
+
+  feature("Get Cases sorted by commodity code") {
+
+    val caseWithEmptyCommCode = createCase().copy(decision = None)
+    val caseY1 = createCase().copy(decision = Some(createDecision(bindingCommodityCode = "777")))
+    val caseY2 = createCase().copy(decision = Some(createDecision(bindingCommodityCode = "777")))
+    val caseZ = createCase().copy(decision = Some(createDecision(bindingCommodityCode = "1111111111")))
+
+    scenario("Sorting default - ascending order") {
+      Given("There are few cases in the database")
+      storeCases(caseY2, caseWithEmptyCommCode, caseY1, caseZ)
+
+      When("I get all cases sorted by commodity code")
+      val result = Http(s"$serviceUrl/cases?sort_by=commodity-code").asString
+
+      Then("The response code should be 200")
+      result.code shouldEqual OK
+
+      And("The expected cases are returned in the JSON response")
+      Json.parse(result.body) shouldBe Json.toJson(Seq(caseWithEmptyCommCode, caseZ, caseY2, caseY1))
```

Comment:
are we happy with this order? 
the cases with no ruling (no decision) come first in the results

Reply:
I think so, they will be filtered out as part of my current story/task